### ISSUE TITLE
Does this game have a memory leak?

### ISSUE THREAD

**Issue by samreid** [phetsims/build-an-atom#99]
I am seeing lots of listeners added, properties created, etc without any dispose()

**Comment by samreid** [comment]
By the way, I'm adding a lot of dispose functions for compatibility with together.

**Comment by samreid** [comment]
Running the fuzz tester on the published version look like this:

![image](https://cloud.githubusercontent.com/assets/679486/6986752/a9502f50-d9fd-11e4-818c-7ff5e2912a26.png)

So without any unlinks, or disposes, why is there no memory leak apparent?

**Comment by aadish** [comment]
dispose functions have been added to tackle memory leak, assigning to @jbphet to review

**Comment by jbphet** [comment]
For the most part these changes look good.  There were a handful of places where an anonymous update function was consolidated into something that didn't really convey what it does, for instance, one of the functions was named ```depiction```.  Please review the names of the functions and make it is clear what it does, even if it's just changing the name to something like ```updateDepiction```.

**Comment by jbphet** [comment]
@aadish and I did a lot of profiling on this after the changes that he submitted above.  In order to eliminate leaks from the other screens and from the framework, we ran with the ```screens=3``` query parameter.  With this, we saw a pattern where the heap memory would increase for the first several minutes of fuzz testing, then would go up an down, maxing out at about 17.7 MB.  This leads me to conclude that any leaks have been addressed.
